### ISSUE TITLE
Random noise on display

### ISSUE THREAD

**Issue by Jackz3885** [TaraHoleInIt/tarablessd1306#9]
When compiling and using the example HelloWorld, the text displays fine however it is turning on other random pixels which are not present when using the Arduinolibraries

**Comment by TaraHoleInIt** [comment]
Which display are you using?

**Comment by Grtschnk** [comment]
Having the same issue. The library seems to be working, but there are random pixels. Using other libraries resets the noise and it is gone for a while. Until I turn it back on the next day. My display should be a 4-pin SSD1306.

**Comment by TaraHoleInIt** [comment]
Is it I2C or SPI?
There are a lot of those SSD1306 displays out there and not all of them are the same, it's possible my code doesn't play nice with some of them.

Can you post a link to the one you bought?

**Comment by Grtschnk** [comment]
Yes, forgot the mention it is I2C. I will have a look if I can find the link.
On the back it says: GM009605

I will now try if SPI works/changes anything. 

Thank you for your library and your quick reply!

Did find my old order confirmation, however the article page is not accesible anymore. This one looks close: https://www.aliexpress.com/item/4000790237487.html?spm=a2g0o.productlist.0.0.8a11508a5WtSpH&algo_pvid=f52a9edb-d4eb-477c-b191-033c54e6b776&algo_expid=f52a9edb-d4eb-477c-b191-033c54e6b776-0&btsid=0be3764515882959420161558efe02&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_

**Comment by Grtschnk** [comment]
Jokes on me. SPI needs more pins. I only have VCC, GND, SCK and SDA

**Comment by doragasu** [comment]
I have the same problem. Random noise on the HelloWorld project. The weird thing is the other examples (snow, fonts) work great, but I am having also problem with my own application.

Display is SSD1306 connected using SPI. I have tested lowering speed to 40 kHz and introducing small delays without luck.

**Comment by doragasu** [comment]
OK, found my problem, hope it is the same for other users. I did not change the default 128x32 px in the HelloWorld example. Setting it to the correct 128x64 corresponding to my display, made the problem go away, now text is rendered beautifully.

Great library BTW, thanks!

**Comment by pearson** [comment]
I had exactly the same issue. At first I'd wondered whether it was intentional and related to the "snow" example. 😄 

Thanks for this library, @TaraHoleInIt !
